cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -87,185 +90,213 @@ public void addStateVariablesToCgmes() {
 
         addVoltagesForBoundaryNodes();
 
-        addPowerFlowToCgmes();
+        addPowerFlows();
 
-        addShuntCompensatorSectionsToCgmes();
+        addShuntCompensatorSections();
 
-        addTapStepToCgmes();
+        addTapSteps();
 
-        addStatusToCgmes();
+        addStatus();
     }
 
     private void addVoltagesForTopologicalNodes() {
-        PropertyBags voltages = new PropertyBags();
-        // add voltages for TpNodes existing in the Model
-        for (PropertyBag tn : cgmes.topologicalNodes()) {
-            Bus b = network.getBusBreakerView().getBus(tn.getId(CgmesNames.TOPOLOGICAL_NODE));
-            PropertyBag p = new PropertyBag(SV_VOLTAGE_PROPERTIES);
+        // Try to add voltages for all TP nodes existing in the CGMES model
+        PropertyBags svVoltages = new PropertyBags();
+        if (cgmes.isNodeBreaker()) {
+            addVoltagesForTopologicalNodesNodeBreaker(svVoltages);
+        } else {
+            // For CGMES bus-branch models:
+            // In IIDM there is a bus at bus-breaker view
+            // for each CGMES topological node
+            cgmes.topologicalNodes().stream().forEach(tp -> {
+                String tpId = tp.getId(CgmesNames.TOPOLOGICAL_NODE);
+                Bus b = network.getBusBreakerView().getBus(tpId);
+                if (b != null) {
+                    svVoltages.add(buildSvVoltage(tp.getId(CgmesNames.TOPOLOGICAL_NODE), b.getV(), b.getAngle()));
+                } else {
+                    // If no IIDM bus has been found it may correspond
+                    // to a boundary, that will be added later
+                    // FIXME(Luma) We should have uniform processing of TP nodes
+                    // present in the original data that have not received an SvVoltage
+                    // this processing should be common to node-breaker and bus-branch
```

Comment:
I think that @zamarrenolm is talking about the TN that are located in the middle of boundary lines. They should be updated but we don't have any iidm bus at this location. To update these TN, I think that we have to increase and use the extension ````MergedXnode```` (from enstoe-util) but it has to be re-think from in the prism of CGMES use case where we have to store V and angle, but also EquivalentInjection ids. @zamarrenolm @MioRtia what do you have in mind ?

Reply:
I think that we will not have missing buses from the bus breaker view because we have a bus as soon as we have a branch. But we will met TN without any bus from the node breaker view as soon as in CGMES, we can have a branch connected to a CN without any bus bar section. I don't know for the moment if it is a big issue.